### ISSUE TITLE
Add GitHub Rules & setup for auto-delete branch post PR merge feature

### ISSUE THREAD

**Issue by SID262000** [idodav/sharkio#168]
**Is your feature request related to a problem? Please describe.**
Too many feature branches are showing up as part of the main repo.
**Describe the solution you'd like**
We can simply auto-delete each branch once the PR from it is merged into the main branch
**Describe alternatives you've considered**
No alternative approach exists at the moment, although many prefer to retain the feature branch

**Comment by shravan20** [comment]
@SID262000: This is already added as part of custom repo settings

**Comment by SID262000** [comment]
@shravan20 I had a meeting with @idodav & looks like we don't have it. In case you feel its already in place, feel free to paste a snapshot here, Thanks

**Comment by idodav** [comment]
@SID262000 we have branch protection rules. Do you have any reference to how we can setup auto delete after merge to main?

**Comment by shravan20** [comment]
![image](https://github.com/idodav/sharkio/assets/47430686/cc590929-d329-430c-9d33-337e575c1e14)

This is the PR merged yesterday, which was auto deleted after merge.

**Comment by shravan20** [comment]
> @SID262000 we have branch protection rules. Do you have any reference to how we can setup auto delete after merge to main?

Just in case if this is not happening, @idodav: Adding the [reference](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches)

**Comment by idodav** [comment]
@shravan20 the branch was deleted manually by me. I now activated:
Automatically delete head branches feature. Thanks 😁

**Comment by SID262000** [comment]
@idodav Do we have the rules in place apart from the branch rules setup, yesterday when we explored, we saw it was empty.

**Comment by idodav** [comment]
@SID262000 added this. I'm closing this issue. Thanks.
